app/javascript/hooks/request-query.ts

```diff
@@ -6,30 +6,32 @@ import {
   usePaginatedQuery,
   useQuery,
 } from 'react-query'
-import { UrlParams } from '../utils/url-params'
 import { camelizeKeys } from 'humps'
 import { sendRequest } from '../utils/send-request'
-
-type RequestQuery = ConstructorParameters<typeof UrlParams>[0]
+import { stringify } from 'qs'
 
 export type Request = {
   endpoint: string
-  query?: RequestQuery
+  query?: Record<string, any>
   options: QueryConfig<any>
 }
 
 type PaginatedRequest = {
   endpoint: string
-  query?: RequestQuery
+  query?: Record<string, any>
   options: PaginatedQueryConfig<any>
 }
 
 function handleFetch(
   request: Request,
   isMountedRef: React.MutableRefObject<boolean>
 ) {
+  const params = request.query
+    ? stringify(request.query, { arrayFormat: 'brackets' })
```

Comment:
At first, I tried doing nested objects to params which led me to use `stringify`. We don't need that anymore, but decided to keep it as it's easier for us to use this package rather than maintaining our own.